### DISCUSSION TITLE
A way to revalidate `generateStaticParams`

### DISCUSSION THREAD

**Discussion by gksander** [vercel/next.js#52134] [App Router]
## The Scenario

I have a catch-all route that will render content from a headless CMS, e.g. `/app/[[...slug]]/page.tsx` – which allows content authors in the CMS to create/delete pages on the fly.

Is there any way to use `generateStaticParams` on this route to statically generate these pages _at build time_, and then revalidate this "allowlist" dynamically when a page is created in the CMS? 

According to [the `generateStaticParams` docs](https://nextjs.org/docs/app/api-reference/functions/generate-static-params):

> During revalidation (ISR), generateStaticParams will not be called again.

So it doesn't appear there's a way to do this. 

## The Question

Is there any guidance on how to do this sort of static generation with dynamic routes in a way that allows you to statically generate your site at build time, and then incrementally add to/remove from the built artifacts as CMS entities are created/deleted?

I know there's a `export const dynamicParams = 'true'` option, but then for optional catch-all routes, this allows someone to spam the headless CMS by requesting various non-allowlisted routes that are handled by the catch-all route, so I'd like to stick to `dynamicParams = 'false'` if possible.

**Comment by icyJoseph** [comment]
AFAIK this was also an issue in GetStaticPaths, because you'd need to use fallback, set to true or blocking, to allow runtime modifications. 

It feels like a solution, for this use case, lies outside of Next.js' current API.

Like you'd probably need to rate-limit users, or use a synchronising-DB that copies the CMS' available paths, or a Cache layer that limits the max amount of requests per unit of time that actually do arrive to the CMS, like max 1 request per 10 seconds, otherwise serve cached content. Or implement a quick way to check if  a path exists, and if it doesn't skip the rest of the process.

Or maybe just trigger a rebuild, sounds crazy, but it might just be the easiest thing to do here. Of course a high throughput of changes renders this option useless.

**Reply by builtbylane** [reply]
it would be great if there were a way to trigger `generateStaticParams` and pass in a specific param to generate.
